### PR TITLE
Core/Scripting: Fix Exile's Reach Alliance Quest "Warming Up"

### PR DESCRIPTION
**Pull request by Seyden**
**Changes proposed:**

-  
-  
-  

**Issues addressed:**
Depends on #27785

Closes #  (insert issue tracker number)

**Tests performed:**

(Does it build, tested in-game, etc.)

**Known issues and TODO list:** (add/remove lines as needed)

- [ ] 
- [ ]

### CODE REVIEW

File: sql/updates/world/master/2022_02_07_03_world_XX.sql [hunk 1]
```diff
@@ -0,0 +1,31 @@
+SET @GARRICK = 1568280;
+SET @GARRICKGUID = -1050145;
+
+-- Captain Garrick (Remove SmartAI and add ScriptName)
+UPDATE creature_template SET AIName = '', ScriptName = 'npc_captain_garrick' WHERE entry = @GARRICK;
```
Comment (mdX7) [inline sql/updates/world/master/2022_02_07_03_world_XX.sql:5]: missing backticks for all fields & tables

File: sql/updates/world/master/2022_02_07_03_world_XX.sql [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+DELETE FROM smart_scripts WHERE entryorguid = @GARRICKGUID;
```
Comment (mdX7) [inline sql/updates/world/master/2022_02_07_03_world_XX.sql:6]: also check `source_type`, you are currently also deleting gameobjects, areatrigger, timedactionlists, ... with that entryorguid

File: src/server/scripts/EasternKingdoms/zone_north_sea.cpp [hunk 3]
```diff
@@ -0,0 +1,119 @@
...
+
+        SetPauseTimer(1ms);
+        me->RemoveNpcFlag(UNIT_NPC_FLAG_QUESTGIVER);
+    }
+
+    void IsSummonedBy(WorldObject* /*owner*/) override
+    {
+        _scheduler.Schedule(1600ms, [this](TaskContext task)
+        {
+            Talk(0);
```
Comment (mdX7) [inline src/server/scripts/EasternKingdoms/zone_north_sea.cpp:82]: magic number

Comment (Seyden) [reply src/server/scripts/EasternKingdoms/zone_north_sea.cpp:82]: its not like TALK_0 or 0 will make any difference here tbh

Comment (Ovahlord) [reply src/server/scripts/EasternKingdoms/zone_north_sea.cpp:82]: doesn't matter. magic numbers are magic numbers. get rid of it

**Comment by MaxtorCoder** [comment]
Why the hell is this script in EasternKingdoms and not in an ExilesReach file..

**Comment by Seyden** [comment]
> Why the hell is this script in EasternKingdoms and not in an ExilesReach file..

Aokromes and me decided to put it where Isle of Quel'Danas is.. because Quel'danas is in the North sea and the North Sea is between Northend and Eastern Kingdom.. so we put it in Eastern Kingdom.. besides.. i won't create a whole new folder just for 1 cpp, doesnt make sense tbh.. Exile's Reach is probably not even big enough for it to be worth

**Comment by MaxtorCoder** [comment]
> > Why the hell is this script in EasternKingdoms and not in an ExilesReach file..
> 
> Aokromes and me decided to put it where Isle of Quel'Danas is.. because Quel'danas is in the North sea and the North Sea is between Northend and Eastern Kingdom.. so we put it in Eastern Kingdom.. besides.. i won't create a whole new folder just for 1 cpp, doesnt make sense tbh.. Exile's Reach is probably not even big enough for it to be worth

It is 😄 Make a Shadowlands folder and do zone_exilesreach.cpp, this is just useless honestly. Following this you will get Exile's Reach content in different script files.

**Comment by Fabi** [comment]
> > > Why the hell is this script in EasternKingdoms and not in an ExilesReach file..
> > 
> > 
> > Aokromes and me decided to put it where Isle of Quel'Danas is.. because Quel'danas is in the North sea and the North Sea is between Northend and Eastern Kingdom.. so we put it in Eastern Kingdom.. besides.. i won't create a whole new folder just for 1 cpp, doesnt make sense tbh.. Exile's Reach is probably not even big enough for it to be worth
> 
> It is 😄 Make a Shadowlands folder and do zone_exilesreach.cpp, this is just useless honestly. Following this you will get Exile's Reach content in different script files.

technically its bfa content. What about just following map.db2 info? it didn't get a parent map so leave it alone :D

File: src/server/scripts/EasternKingdoms/zone_north_sea.cpp (hunk 3 above)
Comment (Fabi) [reply src/server/scripts/EasternKingdoms/zone_north_sea.cpp:82]: > doesn't matter. magic numbers are magic numbers. get rid of it

these names for TALK_xx are not even a bit better than writing 0, 1, ... kek.

**Pull request closed without merging**